### ISSUE TITLE
v0.14.0 - Getting `strip failed with exit code 1` on Archive.

### ISSUE THREAD

**Issue by joshuapinter** [realm/realm-js#503]
I wasn't getting this before but perhaps an issue since 0.14.0.

![screenshot 2016-06-23 07 07 10](https://cloud.githubusercontent.com/assets/180819/16304141/5a9928a2-3911-11e6-9b68-ea3d42c6faa4.png)

I've got all my strip build settings to no. Any idea what could be the issue? 

Thanks!

Josh

**Comment by dorongutman** [comment]
Getting the same thing:

```
Strip /Users/doron/Library/Developer/Xcode/DerivedData/MyAppMobile-aluqrtvwwettmshhhguwuycjsecp/Build/Intermediates/ArchiveIntermediates/MyAppMobile/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libRealmJS.a
    cd /Users/doron/repos/myapp-mobile/node_modules/realm/src
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -S /Users/doron/Library/Developer/Xcode/DerivedData/MyAppMobile-aluqrtvwwettmshhhguwuycjsecp/Build/Intermediates/ArchiveIntermediates/MyAppMobile/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libRealmJS.a

0  0x10ee1597e  __assert_rtn + 144
1  0x10ee6ca06  ld::tool::SectionRelocationsAtom<arm64>::encodeSectionReloc(ld::Internal::FinalSection*, ld::tool::SectionRelocationsAtom<arm64>::Entry const&, std::__1::vector<macho_relocation_info<Pointer64<LittleEndian> >, std::__1::allocator<macho_relocation_info<Pointer64<LittleEndian> > > >&) + 1618
2  0x10ee9732a  ld::tool::SectionRelocationsAtom<arm64>::encode() + 62
3  0x10ee7276d  ld::tool::OutputFile::updateLINKEDITAddresses(ld::Internal&) + 413
4  0x10ee6ce89  ld::tool::OutputFile::write(ld::Internal&) + 167
5  0x10ee168fa  main + 1311
6  0x7fff98cac5ad  start + 1
A linker snapshot was created at:
	/tmp/strip.QeenV8-2016-05-23-171558.ld-snapshot
ld: Assertion failed: (0 && "need to handle arm64 -r reloc"), function encodeSectionReloc, file /Library/Caches/com.apple.xbs/Sources/ld64/ld64-264.3.101/src/ld/LinkEditClassic.hpp, line 1907.
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip: internal link edit command failed
Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip failed with exit code 1
```

The `/tmp/strip.QeenV8-2016-05-23-171558.ld-snapshot` directory has the following relevant file:
`assert_info`:
```
Assertion failed: (0 && "need to handle arm64 -r reloc"), function encodeSectionReloc, file /Library/Caches/com.apple.xbs/Sources/ld64/ld64-264.3.101/src/ld/LinkEditClassic.hpp, line 1907.
0  0x10ee1597e  __assert_rtn + 144
1  0x10ee6ca06  ld::tool::SectionRelocationsAtom<arm64>::encodeSectionReloc(ld::Internal::FinalSection*, ld::tool::SectionRelocationsAtom<arm64>::Entry const&, std::__1::vector<macho_relocation_info<Pointer64<LittleEndian> >, std::__1::allocator<macho_relocation_info<Pointer64<LittleEndian> > > >&) + 1618
2  0x10ee9732a  ld::tool::SectionRelocationsAtom<arm64>::encode() + 62
3  0x10ee7276d  ld::tool::OutputFile::updateLINKEDITAddresses(ld::Internal&) + 413
4  0x10ee6ce89  ld::tool::OutputFile::write(ld::Internal&) + 167
5  0x10ee168fa  main + 1311
6  0x7fff98cac5ad  start + 1
```

**Comment by appden** [comment]
@joshuapinter @dorongutman Please try cleaning your build files (*Product* → *Clean* in the menubar, or ⇧⌘K), and then rebuilding. Also, if that doesn't work, please let me know which version of Xcode you're using. Thanks!

**Comment by dorongutman** [comment]
@appden I'm always doing these 2 actions when such issue arise, and I did them here as well:
1. Cmd+K
1. Clearing the entire DerivedData folder when such issues arise

None of these actions helped.

My xcode is Version 7.3 (7D175)

**Comment by appden** [comment]
@dorongutman That definitely appears to be a bug in `strip`. If you can, please try upgrading to Xcode 7.3.2 and let me know how that goes. I'm not experiencing that issue on my iPhone 6s device on 7.3.2.

**Comment by joshuapinter** [comment]
@appden 7.3.2? The latest I can see is 7.3.1: https://itunes.apple.com/ca/app/xcode/id497799835?mt=12

**Comment by joshuapinter** [comment]
I upgraded to Xcode 7.3.1, did a clean, build and archive and still getting the same error.

**Comment by joshuapinter** [comment]
Just confirm, I had to resort to `realm` package `0.13.2` in order to get our latest version archiving without hitting this error. So, `0.14.0` is the culprit.

**Comment by appden** [comment]
@joshuapinter Thanks for the update. I'll be digging into this to figure out its cause and hopefully get it fixes quickly for a `0.14.1` bugfix release. Since I am unable to reproduce it here, I would really appreciate knowing more about your environment. So I have some questions below, if you don't mind...

1. Would you post a screenshot of your ** → About This Mac** window
2. Is this reproducing for you in a brand new project?
3. Are you only getting this when you **Archive**, or even for **Debug** builds?

Thanks again!

**Comment by joshuapinter** [comment]
No problem, @appden. Thanks for looking into it!

1. ![screenshot 2016-06-27 14 34 08](https://cloud.githubusercontent.com/assets/180819/16394801/42b24aaa-3c74-11e6-8bcd-571c1a6353e4.png)

2. Haven't tested. Can try and make time for this.

3. Only when I Archive, not for Debugging and generally development.

**Comment by appden** [comment]
@joshuapinter Thanks again for being so quick to respond. Would you mind trying one-by-one disabling the **strip** settings for the `RealmJS` static library target (located under **RealmReact.xcodeproj** ▸ **Libraries** ▸ **RealmJS.xcodeproj**). Listed below:
- `Use Separate Strip`
- `Strip Linked Product`
- `Dead Code Stripping`

If any of them (or combination of them) fixes this issue, please let me know! Here's what it would look like if you disabled them all:

<img width="931" alt="screen shot 2016-06-27 at 3 54 29 pm" src="https://cloud.githubusercontent.com/assets/70904/16398493/a345032a-3c7f-11e6-81b4-b25562253e50.png">

**Comment by maxdjohnson** [comment]
I'm experiencing the same issue when I try to create an archive build. I found that it builds successfully only if the "Strip Linked Product" setting is switched off. Google searching got me nowhere but it looks like the crashing line happens here: https://opensource.apple.com/source/ld64/ld64-224.1/src/ld/LinkEditClassic.hpp in the default block of `encodeSectionReloc`

**Comment by appden** [comment]
@maxdjohnson Thanks! Can you confirm that you turned off **Strip Linked Product** on the `RealmJS` static library target, or did you do on your application target?

**Comment by maxdjohnson** [comment]
I turned it off on the RealmJS static library target

**Comment by appden** [comment]
@dorongutman @joshuapinter @maxdjohnson We just released `0.14.1-rc`. Would you mind updating to that and seeing if this issue is resolved? Thanks so much!

**Comment by joshuapinter** [comment]
@appden I upgraded to `0.14.1-rc` and we're good to go! 👍  

Didn't even have to manually adjust those strip settings on the RealmJS library.

**Comment by appden** [comment]
That's fantastic to hear! Thanks for the confirmation!

**Comment by maxdjohnson** [comment]
👍  works for me!

**Comment by appden** [comment]
We just released the official `0.14.1`. Thanks again for all your help everyone!

**Comment by MasonIngram** [comment]
Working well for me, Thanks lot
